src/stm32/stm32f1.c

```diff
@@ -123,20 +137,22 @@ gpio_peripheral(uint32_t gpio, uint32_t mode, int pullup)
     if (mode == GPIO_INPUT) {
         cfg = pullup ? 0x8 : 0x4;
     } else if (mode == GPIO_OUTPUT) {
-        cfg = STM_OSPEED;
+        cfg = out_spd;
     } else if (mode == (GPIO_OUTPUT | GPIO_OPEN_DRAIN)) {
-        cfg = 0x4 | STM_OSPEED;
+        cfg = 0x4 | out_spd;
     } else if (mode == GPIO_ANALOG) {
         cfg = 0x0;
     } else {
-        if (mode & GPIO_OPEN_DRAIN)
-            // Alternate function with open-drain mode
-            cfg = 0xc | STM_OSPEED;
-        else if (pullup > 0)
-            // Alternate function input pins use GPIO_INPUT mode on the stm32f1
+        // Alternate function input pins use GPIO_INPUT mode on the stm32f1
+        // so only support AF_PP or AF_OD here
+        if(!(mode & (GPIO_OUTPUT | GPIO_OPEN_DRAIN))) {
             cfg = 0x8;
-        else
-            cfg = 0x8 | STM_OSPEED;
+        } else {
+            if (mode & GPIO_OPEN_DRAIN)
+                cfg = 0xc | out_spd;
+            else
+                cfg = 0x8 | out_spd;
+        }
```

Comment:
I think for my current situation, yep, this would indeed work :)
So that seems like a reasonable fallback.
Tomorrow night I'll have a look at how far the current gpio_peripheral(..) stuff goes (and hence how far the impact of such a change might be..  and I'll chase down the STMicro XL type macros

Reply:
It didn't look too bad, since the flags are isolated to the STM32 targets.
But not strictly required for this, so I'll go with your suggestion of leveraging the GPIO_HIGH_SPEED flag for this.